src/renderers/webvr/WebXRManager.js

```diff
@@ -101,6 +105,11 @@ function WebXRManager( renderer ) {
 
 		referenceSpace = value;
 
+		currentPixelRatio = renderer.getPixelRatio();
+		renderer.getSize( currentSize );
+
+		renderer.setDrawingBufferSize( session.baseLayer.framebufferWidth, session.baseLayer.framebufferHeight, 1 );
```

Comment:
This has a race condition. setSession uses updateRenderState to set the baseLayer, but that only takes effect on the next frame. If the reference space request resolves before that, session.baseLayer is still null at this point.

Conversely, resolving a reference space might take several frames, and the drawing buffer size wouldn't be up to date if that happens.

(It's also legal for an application to switch the baseLayer to one with a different resolution while a session is active, though three.js wouldn't be doing that in the current implementation.)

I think it would be safer to update the size in onAnimationFrame if it has changed.

Thank you for making this change BTW!

Reply:
Yeah, that makes sense. Let me look into later. Sorry I'm a bit busy now.

Reply:
Also, the property was changed to `session.renderState.baseLayer` from `session.baseLayer`, so the error I got was likely due to that, though I think the race condition potentially still exists.

Reply:
Yes, I need to update to follow the latest spec because this PR was first made while ago. Thanks.

Reply:
I replaced `session.baseLayer` with `session.renderState.baseLayer`.

Regarding the race condition. We and this PR currently do in order

1. call session.updateRenderState()
2. reference space request is resolved 
3. update renderer drawing buffer from session.renderState
4. sessionstart event is fired and EffectComposer updates its size from renderer drawing buffer size
5. start session animation

Then no race condition but updating renderer size should always fail...

I opened #17737 for firing the start event in the first animation frame. If it's approved I update the renderer size in the first animation frame before sessionstart event is fired

1. call session.updateRenderState()
2. reference space request is resolved 
3. start session animation
4. in the first animation frame (a.) update renderer drawing buffer from session.renderState
  and (b.) sessionstart event is fired and EffectComposer updates its size from renderer drawing buffer size